### PR TITLE
Add fps and duration to GetCodecInfo

### PR DESCRIPTION
**Pull request by ad-astra-video**
Add FPS and Duration data to GetCodecInfo.

### CODE REVIEW

File: ffmpeg/extras.c [hunk 1]
```diff
@@ -176,6 +176,10 @@ int lpms_get_codec_info(char *fname, pcodec_info out)
       }
       out->width  = ic->streams[vstream]->codecpar->width;
       out->height = ic->streams[vstream]->codecpar->height;
+      
+      AVRational frame_rate = av_guess_frame_rate(ic,ic->streams[vstream],NULL);
+      out->fps = (frame_rate.num && frame_rate.den ? av_q2d(frame_rate) : 0);
```
Comment (leszko) [inline ffmpeg/extras.c:181]: What does this line do? Why isn't it just this?
```
out->fps = frame_rate.num;
```

File: ffmpeg/extras.c [hunk 2]
```diff
@@ -176,6 +176,10 @@ int lpms_get_codec_info(char *fname, pcodec_info out)
       }
       out->width  = ic->streams[vstream]->codecpar->width;
       out->height = ic->streams[vstream]->codecpar->height;
+      
+      AVRational frame_rate = av_guess_frame_rate(ic,ic->streams[vstream],NULL);
```
Comment (leszko) [inline ffmpeg/extras.c:180]: Just to make sure, did you check the performance of this call? Won't it slow down the input file analysis?

**Review by leszko — COMMENTED** [review]
Nice! I tested it and it worked fine for every video I tried! 🎉 

Added two comments just for me to understand the changes.

File: ffmpeg/extras.c (hunk 1 above)
Comment (ad-astra-video) [reply ffmpeg/extras.c]: frame rates are in `AVRational` types that are similar to Rats in Go with a numerator and denominator.  I think this is overly conservative after researching a bit more and changing to `avg_frame_rate` in the AVStream

File: ffmpeg/extras.c (hunk 2 above)
Comment (ad-astra-video) [reply ffmpeg/extras.c]: I have updated to use `avg_frame_rate` in the `AVStream` from the `avformat_find_stream_info` call already being done.  Duration is in the `AVFormatContext` as well so don't need to calculate.  Want to note that the `nb_frames` was not populated on webm files  I think this is a container spec driven difference compared to mp4 files.

**Comment by ad-astra-video** [comment]
Performance stats indicate it performs more or less the same as not getting the FPS and Duration data.  Was tested on my O node so not entirely unused test environment.

Most test files from https://test-videos.co.uk

[perf_test.txt](https://github.com/livepeer/lpms/files/13247245/perf_test.txt)

**Review by leszko — APPROVED** [review]
Looks good. @ad-astra-video ping me when the PR is ready to merge, then I'll merge it.

**Comment by leszko** [comment]
@ad-astra-video Is this PR ready to merge?

**Comment by ad-astra-video** [comment]
I think the lpms part has no changes needed.  go-livepeer does not consider getting the fps from the new fields yet though.  Do you want to wait to merge until that gets added to go-livepeer?

**Comment by leszko** [comment]
> I think the lpms part has no changes needed. go-livepeer does not consider getting the fps from the new fields yet though. Do you want to wait to merge until that gets added to go-livepeer?

Yeah, maybe let's wait and merge it all together

**Comment by ad-astra-video** [comment]
Rebased this to current master.  Testing this with AI workflows.

**Comment by j0sh** [comment]
Is this PR still needed now that https://github.com/livepeer/lpms/pull/407 is merged?

**Pull request closed without merging**